src/party_menu.c

```diff
@@ -5170,13 +5165,26 @@ static void PartyMenuTryEvolution(u8 taskId)
 
     if (targetSpecies != SPECIES_NONE)
     {
-        FreePartyPointers();
```

Comment:
Removing this line is causing it to crash when evolving a mon and then attempting to evolve another. I'd add it back in like this:
```diff
if (targetSpecies != SPECIES_NONE)
{
+   FreePartyPointers();
    if (gSpecialVar_ItemId == ITEM_RARE_CANDY && gPartyMenu.menuType == PARTY_MENU_TYPE_FIELD && CheckBagHasItem(gSpecialVar_ItemId, 1))
        gCB2_AfterEvolution = CB2_ReturnToPartyMenuUsingRareCandy;
    else
        gCB2_AfterEvolution = gPartyMenu.exitCallback;
}
```